src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -2863,13 +2851,12 @@ int FlexPA::getFlatIdx(const int idx_1, const int idx_2, const int idx_2_dim)
 }
 
 // get idx_1 and idx_2 from flat index
-void FlexPA::getNestedIdx(const int flat_idx,
-                          int& idx_1,
-                          int& idx_2,
-                          const int idx_2_dim)
+std::pair<int, int> FlexPA::getNestedIdx(const int flat_idx,
+                                         const int idx_2_dim)
 {
-  idx_1 = flat_idx / idx_2_dim - 1;
-  idx_2 = flat_idx % idx_2_dim;
+  const int idx_1 = flat_idx / idx_2_dim - 1;
+  const int idx_2 = flat_idx % idx_2_dim;
+  return {idx_1, idx_2};
```

Comment:
Any idea where there is a '-1' here?  It seems non-idiomatic.

Reply:
Yes. This function is the reciprocal to getFlatIdx:
```
int FlexPA::getFlatIdx(const int idx_1, const int idx_2, const int idx_2_dim)
{
  return ((idx_1 + 1) * idx_2_dim + idx_2);
}
```
The `-1` inverts the `+1` in the `idx_1`. The `-1`, in turn, is there because the minimal value for `idx_1` is `-1`, not `0`. So the returned value will be always positive. 

`-1` is used as an argument for this function at `genPatterns_reset`, `genPatternsInit`and `genInstRowPatternInit`

Reply:
I actually don't like how the code keeps flattening and nesting these indexes, it happens a lot and some times one right after the other. If I find a way to eliminate it and keep a single representation for these indexes I will implement it.